File: src/ru/sadv1r/afc/ideaPlugin/achievements/HelloWorld.java
package ru.sadv1r.afc.ideaPlugin.achievements;

import com.intellij.execution.filters.Filter;
import com.intellij.notification.Notification;
import com.intellij.notification.NotificationType;
import com.intellij.notification.Notifications;
import com.intellij.openapi.components.ServiceManager;
import org.jetbrains.annotations.Nullable;
import ru.sadv1r.afc.ideaPlugin.Achievable;
import ru.sadv1r.afc.ideaPlugin.Stats;

/**
 * Created by sadvr on 12/16/15.
 */
public class HelloWorld implements Filter, Achievable {
    private final static Stats stats = ServiceManager.getService(Stats.class);
    private int finish = 1;
    private final String name = "Say hi to all";
    private final String text = "Type traditional Hello World";

    @Nullable
    @Override
    public Result applyFilter(String line, int entireLength) {
        if (line.contains("Hello World") && stats.getHelloWorldAchieved() != 1) {
            Notifications.Bus.notify(new Notification("Achievement", getName(), getText(), NotificationType.INFORMATION));

            stats.setHelloWorldAchieved();
        }
        return null;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public String getText() {
        return text;
    }

    @Override
    public int getProgress() {
        return stats.getHelloWorldAchieved();
    }

    @Override
    public int getFinish() {
        return finish;
    }
}

File: src/main/kotlin/ru/sadv1r/afc/ideaPlugin/achievements/HelloWorld.kt
package ru.sadv1r.afc.ideaPlugin.achievements

import ru.sadv1r.afc.ideaPlugin.Achievable
import com.intellij.execution.filters.Filter
import com.intellij.notification.Notification
import com.intellij.notification.NotificationType
import com.intellij.notification.Notifications

class HelloWorld(
    name: String = "Say hi to all",
    text: String = "Type traditional Hello World",
    progress: Int = 0,
    finish: Int = 1
) : Filter, Achievable(name, text, progress, finish) {
    override val progress: Int
        get() = stats.helloWorldAchieved

    override fun applyFilter(line: String, entireLength: Int): Filter.Result? {
        if (line.contains("Hello World") && stats.helloWorldAchieved != 1) { //TODO Добавить механизм отслеживания было ли уже выдано достижение ранее. Сейчас != 1 это костыль
            Notifications.Bus.notify( Notification("Achievement", name, text, NotificationType.INFORMATION))

            stats.setHelloWorldAchieved()
        }
        return null
    }
}